lib/widgets/drive.coffee

```diff
@@ -51,18 +68,45 @@ getDrives = ->
 # visuals.drive('Please select a drive').then (drive) ->
 # 	console.log(drive)
 ###
+
 module.exports = (message = 'Select a drive') ->
 	getDrives().then (drives) ->
-
-		if _.isEmpty(drives)
-			throw new Error('No available drives')
-
-		form.ask
-			type: 'list'
-			name: 'drive'
+		options =
 			message: message
+			name: 'drive'
 			choices: _.map drives, (drive) ->
 				return {
 					name: "#{drive.device} (#{drive.size}) - #{drive.description}"
 					value: drive.device
+					disabled: undefined
 				}
+
+		ui = new UI
+			input: process.stdin
+			output: process.stdout
+		list = new InquirerList(options, ui.rl)
+		scanner = new DriveScanner getDrives,
+			interval: 4000
+			drives: drives
+
+		scanner.on 'add', (drive) ->
+		###
+		New data about drives are automatically being added to both list.opt.choices.choices and
+		list.opt.choices.realChoices through the use of push().
+		https://github.com/SBoudrias/Inquirer.js/blob/master/lib/objects/choices.js#L106
+		###
+			list.opt.choices.push
+				name: "#{drive.device} (#{drive.size}) - #{drive.description}"
+				value: drive.device
+				disabled: undefined
+			list.render()
+
+		scanner.on 'remove', (drive) ->
+			list.opt.choices.choices = cleanupList(list.opt.choices.choices, drive)
+			list.opt.choices.realChoices = cleanupList(list.opt.choices.realChoices, drive)
+			list.render()
+
+		Promise.fromNode (callback) ->
+			list.run (answers) ->
+				return callback(null, answers)
```

Comment:
@jviotti That's weird because the answer always contains the desired result. Also see [this](https://github.com/SBoudrias/Inquirer.js/blob/master/examples/list.js#L28)